lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -753,17 +753,18 @@ def parse_configuration_whitelist(name, requirements)
       #         If :subspecs is the only key in the hash, the hash
       #         should be destroyed because it confuses Gem::Dependency.
       #
-      # @return [void]
+      # @return [TrueClass,FalseClass] Returns true if we added new subspecs
       #
       def parse_subspecs(name, requirements)
         options = requirements.last
-        return requirements unless options.is_a?(Hash)
+        return false unless options.is_a?(Hash)
 
         subspecs = options.delete(:subspecs)
         subspecs.each do |ss|
           store_pod("#{name}/#{ss}", *requirements.dup)
         end if subspecs
         requirements.pop if options.empty?
+        !subspecs.nil?
```

Comment:
Any object can be treated like a Boolean in Ruby. Returning the object is the same as returning whether the object is not nil, it's just less code. 

-Samuel E. Giddins

On Apr 6, 2015, at 5:41 AM, Kyle Fuller <notifications@github.com> wrote:

In lib/cocoapods-core/podfile/target_definition.rb:

>  
>          subspecs = options.delete(:subspecs)
>          subspecs.each do |ss|
>            store_pod("#{name}/#{ss}", *requirements.dup)
>          end if subspecs
>          requirements.pop if options.empty?
> +        !subspecs.nil?
subspecs is an array and not a boolean like this method is expected to return.

—
Reply to this email directly or view it on GitHub.